File: 0062-unique-paths/0062-unique-paths.java
class Solution {
    public int uniquePaths(int m, int n) {
        int[] lastRow = new int[n];
        Arrays.fill(lastRow, 1);

        for(int i=m-1; i>=1; i--){
            int[] newRow = new int[n];
            Arrays.fill(newRow, 1);

            for(int j=n-2; j>=0; j--){
                newRow[j] = lastRow[j] + newRow[j+1];
            }

            lastRow = newRow;
        }

        return lastRow[0];

    }
}

File: 0062-unique-paths/0062-unique-paths.py
class Solution(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        grid = [[0 for _ in range(n)] for _ in range(m)]

        for i in range(0, n):
            grid[0][i] = 1

        for j in range(0, m):
            grid[j][0] = 1

        for i in range(1, m):
            for j in range(1, n):
                grid[i][j] = grid[i-1][j] + grid[i][j-1]

        return grid[m-1][n-1]

            